plugins/coverimage.koplugin/main.lua

```diff
@@ -67,9 +71,40 @@ end
 
 function CoverImage:createCoverImage(doc_settings)
    if self.enabled and not doc_settings:readSetting("exclude_cover_image") == true then
-        local image = self.ui.document:getCoverPageImage()
-        if image then
+        local cover_image = self.ui.document:getCoverPageImage()
+        if cover_image then
+            local s_w, s_h = Device.screen:getWidth(), Device.screen:getHeight()
+            local i_w, i_h = cover_image:getWidth(), cover_image:getHeight()
+            local scale_factor = math.min(s_w / i_w, s_h / i_h)
+
+            if scale_factor == 1 or self.cover_image_background == "none" then
+                cover_image:writePNG(self.cover_image_path, false)
+                return
+            end
+
+            local scaled_w, scaled_h = i_w * scale_factor, i_h * scale_factor
+            local cover_image = RenderImage:scaleBlitBuffer(cover_image, scaled_w, scaled_h)
+
+            -- new buffer with screen dimensions,
+            local image = Blitbuffer.new( s_w, s_h, cover_image:getType() )
+            logger.err("xxxxxxxxxxxxxxxxxxxxxxxx " .. self.cover_image_background)
+            if self.cover_image_background == "black" then
+                image:fill(Blitbuffer.COLOR_BLACK)
```

Comment:
BBs are zero-initialized, so the explicit zero-fill is technically redundant.

(I'd still keep it just to avoid aking the code more puzzling in that case, this is just me throwing in a "fun fact" here ^^).